python-frontend/src/main/java/org/sonar/python/index/DescriptorUtils.java

```diff
@@ -107,27 +114,89 @@ public static AmbiguousDescriptor ambiguousDescriptor(AmbiguousSymbol ambiguousS
+    if (createdSymbols.containsKey(descriptor.fullyQualifiedName())) {
+      return createdSymbols.get(descriptor.fullyQualifiedName());
+    }
     String symbolName = localSymbolName != null ? localSymbolName : descriptor.name();
     switch (descriptor.kind()) {
       case CLASS:
-        return new ClassSymbolImpl((ClassDescriptor) descriptor, projectLevelSymbolTable, symbolName);
+        return createClassSymbol(descriptor, projectLevelSymbolTable, createdSymbols, symbolName);
       case FUNCTION:
-        return new FunctionSymbolImpl((FunctionDescriptor) descriptor, projectLevelSymbolTable, symbolName);
+        return createFunctionSymbol((FunctionDescriptor) descriptor, projectLevelSymbolTable, createdSymbols, symbolName);
       case VARIABLE:
         return new SymbolImpl(symbolName, descriptor.fullyQualifiedName());
       case AMBIGUOUS:
         Set<Symbol> alternatives = ((AmbiguousDescriptor) descriptor).alternatives().stream()
-          .map(a -> DescriptorUtils.symbolFromDescriptor(a, projectLevelSymbolTable, symbolName))
+          .map(a -> DescriptorUtils.symbolFromDescriptor(a, projectLevelSymbolTable, symbolName, createdSymbols))
           .collect(Collectors.toSet());
         return new AmbiguousSymbolImpl(symbolName, descriptor.fullyQualifiedName(), alternatives);
       default:
         throw new IllegalStateException(String.format("Error while creating a Symbol from a Descriptor: Unexpected descriptor kind: %s", descriptor.kind()));
     }
   }
+
+  private static ClassSymbolImpl createClassSymbol(Descriptor descriptor, ProjectLevelSymbolTable projectLevelSymbolTable, Map<String, Symbol> createdSymbols, String symbolName) {
+    ClassDescriptor classDescriptor = (ClassDescriptor) descriptor;
+    ClassSymbolImpl classSymbol = new ClassSymbolImpl((ClassDescriptor) descriptor, symbolName);
+    createdSymbols.put(descriptor.fullyQualifiedName(), classSymbol);
+    addSuperClasses(classSymbol, classDescriptor, projectLevelSymbolTable, createdSymbols);
+    addMembers(classSymbol, classDescriptor, projectLevelSymbolTable, createdSymbols);
+    return classSymbol;
+  }
+
+  private static void addMembers(ClassSymbolImpl classSymbol, ClassDescriptor classDescriptor,
+    ProjectLevelSymbolTable projectLevelSymbolTable, Map<String, Symbol> createdSymbols) {
+    classSymbol.addMembers(classDescriptor.members().stream()
+      .map(memberFqn -> DescriptorUtils.symbolFromDescriptor(memberFqn, projectLevelSymbolTable, null, createdSymbols))
+      .map(member -> {
+        if (member instanceof FunctionSymbolImpl) {
+          ((FunctionSymbolImpl) member).setOwner(classSymbol);
+        }
+        return member;
+      })
+      .collect(Collectors.toList()));
+  }
+
+  private static void addSuperClasses(ClassSymbolImpl classSymbol, ClassDescriptor classDescriptor,
+    ProjectLevelSymbolTable projectLevelSymbolTable, Map<String, Symbol> createdSymbols) {
```

Comment:
_Minor_: I prefer adding an empty line to separate parameters from the body of the method